pandas/plotting/_matplotlib/core.py

```diff
@@ -102,13 +105,24 @@ def __init__(
         table=False,
         layout=None,
         include_bool=False,
+        column: Optional[Label] = None,
         **kwds,
     ):
 
         import matplotlib.pyplot as plt
 
         self.data = data
-        self.by = by
+        self.by = [by] if not isinstance(by, list) and by is not None else by
+
+        if self.by:
+            self._grouped_data_size = len(data.groupby(self.by))
+
+        # Assign the rest of columns into self.columns if by is explicitly defined
+        # while column is not, so as to keep the same behaviour with current df.hist
+        if self.by and column is None:
+            self.columns = [col for col in data.columns if col not in self.by]
```

Comment:
Do you need the comprehension here? I think can just assign data.columns, though curious if this is even required

Reply:
yeah, there are used elsewhere e.g. `columns = MultiIndex.from_product([[key], self.columns])` in core and `grouped = self.data.groupby(self.by)[self.columns]` in hist, to get the columns only used after aggregation to form new data to be served for `_iter_data`, so i think maybe it's okay to leave it here?